### ISSUE TITLE
Auto-indent on closing brace

### ISSUE THREAD

**Issue by alexagranov7** [Microsoft/vscode#67678]
![image](https://user-images.githubusercontent.com/28403336/52084629-5500a980-2557-11e9-9175-8a91ef3efaf0.png)

on typing the closing brace. Meaning VSCode should take the current tab indentation size in account and auto-indent the code-block when typing the closing brace.

**Comment by Kenobi-the-2nd** [comment]
Yes, I would very much like to see this feature added as well. Oftentimes I will type in some code, then need to surround it in a loop, if statement, or other block contained by braces. When doing this, I expect that adding the closing brace after the initial brace will indent all code within the braces. I would expect the same behavior when surrounding code with other characters, such as brackets '[]', parentheses '()', or angled brackets '<>'. 

This is a standard feature that is present in other versions of Visual Studio such as Visual Studio Enterprise and even Visual Studio Community. When starting out using Visual Studio Code, I expected this same functionality and was disappointed to learn it was not present. This feature is a must-have and I sincerely hope that it will be added. 

Thank-you very much for your time.

**Comment by rudamaia** [comment]
visual studio for mac has this... Why it still not happening in vscode?

**Comment by gbirchmeier** [comment]
I don't understand why this is still an outstanding issue.

**Comment by eclazi** [comment]
Adding my support to this, would love to see it implemented

**Comment by aiday-mar** [comment]
Hi in order for us to better understand the issue, could you please include a screen recording of the issue you are seeing, as well as include a code sample where this is happening?

**Comment by aiday-mar** [comment]
Hi I see thank you. As a temporary workaround, you may enable format on type or format on save from the settings:

"editor.formatOnSave": true
OR
"editor.formatOnType": true

**Comment by bflora2** [comment]
I've been aggravated by this problem for a long time, but then I found a VSCode extension named "Universal Format on Type" by Crend King that seems to fix the problem perfectly. Woo-Hoo!

**Comment by chrisjaquet** [comment]
Enabling 
```
"editor.formatOnType": true
```
does note trigger a reformat of the code block (or the line containing the closing brace). It only seems to adjust the indentation.

**Comment by lhmouse** [comment]
Is there really no workaround for this? It is rather annoying to remember to fix whitespace like in here:

(https://stackoverflow.com/questions/63199416/how-to-stop-vscode-from-removing-indent-when-closing-curly-braces)
![Image](https://github.com/user-attachments/assets/115a98c1-3c77-4f84-94f7-8cc4711a14a0)
